src/v/storage/log_manager.h

```diff
@@ -168,7 +168,8 @@ class log_manager {
       storage_resources&,
       ss::sharded<features::feature_table>&) noexcept;
 
-    ss::future<ss::shared_ptr<log>> manage(ntp_config);
+    ss::future<ss::shared_ptr<log>>
+      manage(ntp_config, raft::group_id = raft::group_id{});
```

Comment:
two things.

1. it seems odd that the log needs to know its raft group-id. but i could buy that its needed because we have raft group-id prefixes in kvstore keys, or some other naming that needs it, something like that?
2. if the reason for the raft-group-id is to deal with naming, then is taking a default `group_id{}` here a foot-gun?

Reply:
>needed because we have raft group-id prefixes in kvstore keys, or some other naming that needs it, something like that?

Right, this exactly

>default group_id{} here a foot-gun?

Ah that's a good point, I'll see about removing the default

Reply:
It's pretty widely used in tests. Maybe in a subsequent PR, unless folks feel strongly about it

Reply:
new assertion looks good